lib/Doctrine/ORM/EntityManager.php

```diff
@@ -134,52 +128,47 @@
     /** @var Cache The second level cache regions API. */
     private $cache;
 
+    /** @var MetadataCollection */
+    private $metadatas;
+
     /**
      * Creates a new EntityManager that operates on the given database connection
      * and uses the given Configuration and EventManager implementations.
      *
      */
-    protected function __construct(Connection $conn, Configuration $config, EventManager $eventManager)
+    protected function __construct(Connection $conn, Configuration $config, EventManager $eventManager, MetadataCollection $metadataCollection)
     {
-        $this->conn         = $conn;
-        $this->config       = $config;
-        $this->eventManager = $eventManager;
-
-        $metadataFactoryClassName = $config->getClassMetadataFactoryName();
-
-        $this->metadataFactory = new $metadataFactoryClassName();
-
-        $this->metadataFactory->setEntityManager($this);
-        $this->metadataFactory->setCacheDriver($this->config->getMetadataCacheImpl());
+        $this->conn            = $conn;
+        $this->config          = $config;
+        $this->eventManager    = $eventManager;
+        $this->metadatas       = $metadataCollection;
```

Comment:
`metadata` is irregular: the plural of `metadata` is `metadata`. A better way to name this is probably `mappings`